manifests/forwarder/install.pp

```diff
@@ -64,4 +64,16 @@
     install_options => $splunk::forwarder::install_options,
   }
 
+  if $splunk::forwarder::manage_splunk_user {
+    group { $splunk::forwarder::splunk_user:
+      ensure  => 'present',
+      require => Package[$splunk::forwarder::forwarder_package_name],
+    }
+    user { $splunk::forwarder::splunk_user:
+      ensure  => 'present',
+      home    => $splunk::forwarder::forwarder_homedir,
+      require => Group[$splunk::forwarder::splunk_user],
```

Comment:
Is there a specific reason in this ensures that the splunk package is installed first? I have a use case where I'd prefer to be able to create the splunk user and setup the filesystem under splunk with the proper permissions before installing splunk. For my use case I would have to set manage_splunk_user to false and then manually manage it instead of just taking advantage of this feature. I guess this would cause problems with the home directory not existing?

Reply:
Quick pass through some of the various code paths in the module and it looks as if runuser is only utilized when the service starts, not at any point during the installation process so this at minimum needs to happen before Class[splunk::{enterprise, forwarder}::service] but should be fine happening after package installation...

@nick-markowski Correct me if I missed something.